File: real_time/src/main/scala/util/DateUtil.scala
package util

import java.text.SimpleDateFormat
import java.util.Date

/**
  * Created by Liu HangZhou on 2020/03/29
  * desc: 日期工具类
  */
object DateUtil {

  def main(args: Array[String]): Unit = {
    println(getCurrentDate())
  }

  private val formatDay = new SimpleDateFormat("yyyy-MM-dd")
  private val formatHour = new SimpleDateFormat("yyyy-MM-dd HH")


  //获取当前日期
  def getCurrentDate()={
    val date = new Date()
    val currentDay = formatDay.format(date)
    currentDay
  }


  //将时间戳转换为日期
  def getDate(ts: Long)={
    val date = new Date(ts)
    formatHour.format(date)
  }

}

File: gmall_publisher/src/main/java/com/nyist/gmall/gmall_publisher/util/DateUtil.java
package com.nyist.gmall.gmall_publisher.util;

import org.apache.commons.lang.time.DateUtils;
import sun.applet.Main;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by Liu HangZhou on 2020/03/30
 * desc: 日期工具类
 */
public class DateUtil {

    private static SimpleDateFormat dateSimpledate = new SimpleDateFormat("yyyy-MM-dd");


    //获取指定日期的前一天
    public static String getYesterday(String dateString) {

        Date date = null;
        try {
            date = dateSimpledate.parse(dateString);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        Date yesterDay = DateUtils.addDays(date, -1);
        String yesterday = dateSimpledate.format(yesterDay);
        return yesterday;
    }


    public static void main(String[] args){

        System.out.println(getYesterday("2020-03-30"));
    }




}
